lib/coarsecoarse_op.cu

```diff
@@ -107,8 +107,16 @@ namespace quda {
 #endif
     } else if (coarseColor == 24) {
       calculateYcoarse<Float,vFloat,fineColor,fineSpin,24,coarseSpin>(Y, X, Yatomic, Xatomic, uv, T, g, clover, cloverInv, kappa, mu, mu_factor, dirac, matpc);
+#ifdef GPU_WILSON_DIRAC
```

Comment:
Do you need 24 for staggered as well, or can you extend this guard up to 24 too?

Reply:
Also, shouldn't these `GPU_WILSON_DIRAC` be `NSPIN4`.  E.g., we can compile twisted-mass without Wilson.

Reply:
We need 24 for coarsening the pseudo-fine/KD operator. (This will still be true after adding the optimized KD op, since we'll still explicitly build the coarse links exactly for coarsening, we'll just let managed memory sweep the details under the rug.) Duly noted on `NSPIN4` and `NSPIN1`, fixed.